jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageWriter.java

```diff
@@ -120,8 +120,8 @@ private static void moveIfDoesNotExist(Path source, Path destination) throws IOE
                 () -> {
                   if (Files.exists(destination)) {
                     // If the file already exists, we skip renaming and use the existing file.
-                    // This happens if a new layer happens to have the same content as a
-                    // previously-cached layer.
+                    // This happens, e.g., if a new layer happens to have the same content as a
+                    // previously-cached layer or the same layer is being cached concurrently.
```

Comment:
Does this mean if the same layer is being cached twice?

Reply:
Yes. To be strict, multiple threads may be attempting to cache the same layer more than once (whether almost at the same time or apart). They will all write files at their own temporary location. In the end, all of them will try to rename the temp directory to the final one, which is unique. Then, only one rename will succeed, and other will see that the directory already exists.